pkg/controller/servicebindingrequest/annotations.go

```diff
@@ -0,0 +1,67 @@
+package servicebindingrequest
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+)
+
+const (
+	sbrNamespaceAnnotation = "service-binding-operator.apps.openshift.io/binding-namespace"
+	sbrNameAnnotation      = "service-binding-operator.apps.openshift.io/binding-name"
+)
+
+// extractNamespacedName returns a types.NamespacedName if the required service binding request keys
+// are present in the given data
+func extractNamespacedName(data map[string]string) types.NamespacedName {
+	namespacedName := types.NamespacedName{}
+	ns, exists := data[sbrNamespaceAnnotation]
+	if !exists {
+		return namespacedName
+	}
+	name, exists := data[sbrNameAnnotation]
+	if !exists {
+		return namespacedName
+	}
+	namespacedName.Namespace = ns
+	namespacedName.Name = name
+	return namespacedName
+}
+
+// GetSBRNamespacedNameFromObject returns a types.NamespacedName if the required service binding
+// request annotations are present in the given runtime.Object, empty otherwise. An error can be
+// returned in the case the object can't be decoded.
+func GetSBRNamespacedNameFromObject(obj runtime.Object) (types.NamespacedName, error) {
```

Comment:
Perhaps we can change the signature to return `*types.NamespacedName` instead, since it is expected that no error has happened and the object doesn't refer to any ServiceBindingRequest. This means that `return nil, nil` is a valid return indicating that no error has happened but no value has been found in the given `obj`.

This would influence the `IsSBRNamespaceNameEmpty` method, where a pointer would be used as input and the check for null-ability performed in addition to the existing checks.

Reply:
True, that can be introduced again. I changed it back during the process, but now it's not needed anymore.